core/src/main/java/com/segment/analytics/kotlin/core/Settings.kt

```diff
@@ -29,22 +30,41 @@ data class Settings(
         return typedSettings
     }
 
-    fun isDestinationEnabled(name: String): Boolean {
-        return integrations.containsKey(name)
+    fun hasIntegrationSettings(plugin: DestinationPlugin): Boolean {
+        return hasIntegrationSettings(plugin.key)
+    }
+
+    fun hasIntegrationSettings(key: String): Boolean {
+        return integrations.containsKey(key)
     }
 }
 
 internal fun Analytics.update(settings: Settings, type: Plugin.UpdateType) {
     timeline.applyClosure { plugin ->
         if (plugin is DestinationPlugin) {
-            plugin.enabled = settings.isDestinationEnabled(plugin.key)
+            plugin.enabled = settings.hasIntegrationSettings(plugin)
         }
         // tell all top level plugins to update.
         // For destination plugins they auto-handle propagation to sub-plugins
         plugin.update(settings, type)
     }
 }
 
+fun Analytics.manuallyEnableDestination(plugin: DestinationPlugin) {
+    analyticsScope.launch(analyticsDispatcher) {
+        store.dispatch(
+            System.AddDestinationToSettingsAction(destinationKey = plugin.key) { settings: Settings? ->
```

Comment:
probably don't need the `afterClosure`, since `dispatch` is a suspend function, we can move the `if` block below `store.dispacth`

Reply:
we still need the after closure bcos we need the resultant `settings` from the action

Reply:
do u think this is a better way to do it?
```kotlin
fun Analytics.manuallyEnableDestination(plugin: DestinationPlugin) {
    analyticsScope.launch(analyticsDispatcher) {
        store.dispatch(
            System.AddDestinationToSettingsAction(destinationKey = plugin.key),
            System::class
        )
        val system = store.currentState(System::class)
        system?.settings?.let { settings ->
            findAll(DestinationPlugin::class).forEach {
                plugin.enabled = settings.hasIntegrationSettings(it.key)
            }
        }
    }
}
```

Reply:
yeah. I think this is better, though performance-wise it's a bit slower. one of the benefits of suspend function is avoiding indentation/callback hell. I think we should take it.